purchase_propagate_qty/models/purchase.py

```diff
@@ -20,11 +19,30 @@ def _propagage_qty_to_moves(self):
         for line in self:
             if line.state != "purchase":
                 continue
-            for move in line.move_dest_ids | line.move_ids:
-                if move.state in ("cancel", "done"):
-                    continue
-                if move.product_id != line.product_id:
-                    continue
-                move.product_uom_qty = line.product_uom_qty
-                if float_is_zero(move.product_uom_qty, line.product_uom.rounding):
-                    move._action_cancel()
+            moves = line.move_dest_ids | line.move_ids
```

Comment:
You can not sum qties of `move_dest_ids` and `move_ids`, or you will double the expected qty for sure if there is a downstream move.
I would process them separately, the original implementation was fine I think for downstream moves:
```suggestion
            for move in line.move_dest_ids:
                if move.state in ("cancel", "done"):
                    continue
                if move.product_id != line.product_id:
                    continue
                move.product_uom_qty = line.product_uom_qty
                if float_is_zero(move.product_uom_qty, line.product_uom.rounding):
                    move._action_cancel()
            moves = line.move_ids
```